reconcile/utils/rosa/session.py

```diff
@@ -0,0 +1,182 @@
+import tempfile
+from types import TracebackType
+from typing import Optional, Type
+
+from reconcile.utils.aws_api import (
+    AWSSessionBuilder,
+    AWSStaticCredsSessionBuilder,
+)
+from reconcile.utils.jobcontroller.controller import K8sJobController
+from reconcile.utils.jobcontroller.models import JobConcurrencyPolicy, JobStatus
+from reconcile.utils.ocm_base_client import (
+    OCMAPIClientConfiguration,
+    OCMAPIClientConfigurationProtocol,
+    OCMBaseClient,
+    init_ocm_base_client,
+)
+from reconcile.utils.rosa.model import ROSACluster
+from reconcile.utils.rosa.rosa_cli import (
+    LogHandle,
+    RosaCliException,
+    RosaCliResult,
+    RosaJob,
+)
+from reconcile.utils.secret_reader import SecretReaderBase
+
+
+class RosaSession:
```

Comment:
Is this class intend to be the starting point of all ROSA cli run going forward? Including my use case, the longer script that involves:
1. creation of ocm-role, account-roles, user-role
2.cluster creation, 
3. operator-roles and oidc config creation

They will be in a new method in this class?

Reply:
yes that is the idea. such new methods would then call the `cli_execute` function and pass in their command. as the doc strings state, multiple commands would need to be wrapped into  `bash -c "cmd 1 ; cmd2 ; cmd3"`